### ISSUE TITLE
Hausdorff distance not working

### ISSUE THREAD

**Issue by mateuszbaran** [JuliaImages/ImageDistances.jl#59]
I've tried computing the Hausdorff distance as follows and it errored:
```julia
julia> using ImageDistances

julia> d = Hausdorff()
GenericHausdorff{ImageDistances.MaxReduction,ImageDistances.MaxReduction}(ImageDistances.MaxReduction(), ImageDistances.MaxReduction())

julia> a = rand(Bool, 3, 3)
3×3 Array{Bool,2}:
 0  1  0
 0  1  1
 0  1  1

julia> b = rand(Bool, 3, 3)
3×3 Array{Bool,2}:
 1  0  0
 0  0  1
 1  0  0

julia> d(a, b)
ERROR: MethodError: no method matching (::GenericHausdorff{ImageDistances.MaxReduction,ImageDistances.MaxReduction})(::Float64, ::Float64)
Stacktrace:
 [1] result_type(::GenericHausdorff{ImageDistances.MaxReduction,ImageDistances.MaxReduction}, ::Type{T} where T, ::Type{T} where T) at /home/mateusz/.julia/packages/Distances/nL1rV/src/generic.jl:36
 [2] result_type(::GenericHausdorff{ImageDistances.MaxReduction,ImageDistances.MaxReduction}, ::Array{Float64,2}, ::Array{Float64,2}) at /home/mateusz/.julia/packages/Distances/nL1rV/src/generic.jl:35
 [3] evaluate_hausdorff(::GenericHausdorff{ImageDistances.MaxReduction,ImageDistances.MaxReduction}, ::Array{Bool,2}, ::Array{Bool,2}, ::Array{Float64,2}, ::Array{Float64,2}) at /home/mateusz/.julia/packages/ImageDistances/SDePf/src/hausdorff.jl:102
 [4] (::GenericHausdorff{ImageDistances.MaxReduction,ImageDistances.MaxReduction})(::Array{Bool,2}, ::Array{Bool,2}) at /home/mateusz/.julia/packages/ImageDistances/SDePf/src/hausdorff.jl:128
 [5] top-level scope at REPL[7]:1
```

It seems to work if I define something like this:
```julia
Distances.result_type(::GenericHausdorff, ::Type{T}, ::Type{U}) where {T,U} = promote_type(T, U)
```

Is this a reasonable fix? Or did I do something wrong here?

**Comment by juliohm** [comment]
@jsenn could you take a look at this? Maybe your latest PR has changed some behavior regarding the binary case?

**Comment by johnnychen94** [comment]
Oh yes, I believe so! There was a "breaking" change in upstream package Distances for `result_type`(https://github.com/JuliaStats/Distances.jl/pull/194) so this method is probably needed now.

@mateuszbaran Would you mind providing a patch for this?

**Comment by mateuszbaran** [comment]
Sure, I'm working on a PR that fixes this and the Hausdorff part of #44 .

**Comment by johnnychen94** [comment]
Should be fixed by #60
